main.go

```diff
@@ -64,19 +73,48 @@ func extractSemVersionFromChannel(channel, prefix string) (*semver.Version, erro
 	return semver.NewVersion(trimmed)
 }
 
-// dropPatch returns a semver.Version containing only the major.minor parts.
-func dropPatch(v *semver.Version) (*semver.Version, error) {
-	newVersionStr := fmt.Sprintf("%v.%v", v.Major(), v.Minor())
-	return semver.NewVersion(newVersionStr)
+// splitFastVersion splits the input string into a prefix (including the hyphen)
+// and the version part. It assumes that the input always contains a hyphen.
+func splitChannel(channel string) (string, string, error) {
+	idx := strings.Index(channel, "-")
+	// If the hyphen is not found, return an empty prefix and the original input as version.
+	if idx == -1 {
+		return "", channel, fmt.Errorf("invalid channel format: %s", channel)
+	}
+	prefix := channel[:idx+1]
+	version := channel[idx+1:]
+	return prefix, version, nil
 }
 
 // discoverReleases discovers new releases from the startChannel and minimum acceptable version.
 // It returns a ReleasesByChannel, with keys as full channel names.
-func discoverReleases(client *http.Client, startChannel string, minVer, minChannelVer *semver.Version, prefixes []string, arch string) (ReleasesByChannel, error) {
+func discoverReleases(client *http.Client, startChannels []string, arch string) (ReleasesByChannel, error) {
+	var prefixes []string
+	var minVersion *semver.Version
+	var queue []string
+	queued := map[string]bool{}
+
+	for _, startChannel := range startChannels {
```

Comment:
nit: `startChannels` seems like unnecessary complication, compared to your earlier singular `startChannel`, especially since you set it up from a singular command-line arg with `[]string{*startChannel}`.  But also :shrug:, you've already built it, so maybe not worth dropping the loop now, just in case it turns out to be useful later?

Reply:
Eventually I will move this code to the ClusterService. It really depends on the requirements. For now, I would like to keep it. If it turns out that there is a need to scan only a single channel, I will change the code.